### PR TITLE
Fix for Mnemonic Word Count 9 generation and Unit Test Coverage

### PR DESCRIPTION
**Pull request by safestak-keith**
Implemented simple fix for generating mnemonic for word count of 9 and added initial unit test coverage around these scenarios.

### CODE REVIEW

File: CardanoSharp.Wallet/KeyService.cs [hunk 1]
```diff
@@ -25,20 +25,23 @@ public interface IKeyService
...
-            var entropySize = allowedEntropyLengths[allowedWordLengths.ToList().FindIndex(x => x == wordSize)];
+            if (!allowedWordLengths.Contains(wordSize))
+                throw new ArgumentOutOfRangeException(nameof(wordSize), $"{nameof(wordSize)} must be one of the following values ({string.Join(", ", allowedWordLengths)})");
+
+            var entropySize = allowedEntropyLengths[Array.FindIndex(allowedWordLengths, x => x == wordSize)];
+            if (!allowedEntropyLengths.Contains(entropySize))
+                throw new ArgumentOutOfRangeException(nameof(entropySize), $"Derived entropy {entropySize} is not within the allowed values ({string.Join(", ", allowedEntropyLengths)})");
 
             var rng = new RNGCryptoServiceProvider();
             if (rng is null)
```
Comment (safestak-keith) [inline CardanoSharp.Wallet/KeyService.cs:43]: Unless I am overlooking something in the way RNGCryptoServiceProvider is initialised, the condition if (rng is null) should never be true?

**Review by safestak-keith — COMMENTED** [review]
A few suggestions and points of discussion that I can potentially bundle to this PR following the boy scout rule

File: CardanoSharp.Wallet/KeyService.cs (hunk 1 above)
Comment (nothingalike) [reply CardanoSharp.Wallet/KeyService.cs:43]: yeah i believe thats an old condition, we can probably remove that

File: CardanoSharp.Wallet/KeyService.cs [hunk 2]
```diff
@@ -25,8 +25,8 @@ public interface IKeyService
     public class KeyService : IKeyService
     {
         #region BIP39
-        private readonly int[] allowedEntropyLengths = { 16, 20, 24, 28, 32 };
-        private static readonly int[] allowedWordLengths = { 12, 15, 18, 21, 24 };
+        private readonly int[] allowedEntropyLengths = { 12, 16, 20, 24, 28, 32 };
+        private static readonly int[] allowedWordLengths = { 9, 12, 15, 18, 21, 24 };
         private uint[] wordIndexes;
```
[the remark this answers is not in the document]
Comment (safestak-keith) [reply CardanoSharp.Wallet/KeyService.cs:30]: Created #10 - will resolve this convo

File: CardanoSharp.Wallet/KeyService.cs [hunk 3]
```diff
@@ -25,8 +25,8 @@ public interface IKeyService
...
         private string[] allWords;
```
[the remark this answers is not in the document]
Comment (safestak-keith) [reply CardanoSharp.Wallet/KeyService.cs:31]: Created #12 - resolving this discussion

File: CardanoSharp.Wallet/KeyService.cs [hunk 4]
```diff
@@ -25,8 +25,8 @@ public interface IKeyService
     public class KeyService : IKeyService
     {
         #region BIP39
-        private readonly int[] allowedEntropyLengths = { 16, 20, 24, 28, 32 };
-        private static readonly int[] allowedWordLengths = { 12, 15, 18, 21, 24 };
+        private readonly int[] allowedEntropyLengths = { 12, 16, 20, 24, 28, 32 };
```
[the remark this answers is not in the document]
Comment (nothingalike) [reply CardanoSharp.Wallet/KeyService.cs:28]: Lets make this an Issue/Enhancement

**Review by nothingalike — APPROVED** [review]

**Pull request merged**